Pinta.Core/Pinta.Core.csproj

```diff
@@ -3,6 +3,8 @@
     <TargetFramework>net6.0</TargetFramework>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <Nullable>enable</Nullable>
+    <RuntimeIdentifiers>linux-x64</RuntimeIdentifiers>
```

Comment:
idk? 😕 

I'd rather you tell me

Reply:
I was more asking if there were reasons for adding it in the first place :)
But in the docker file I see `--runtime linux-x64` being used so I'd expect that this does not need to be in the .csproj files unless there is a requirement I'm not aware of

Reply:
Reasons are: it didn't build otherwise (apologies, but no exact error available for now 😕) 

I was thinking it's more like: you have to explicitly define "all" platforms, even if you are building for all

Reply:
The result of the aforementioned patch are:
```
/usr/share/dotnet/sdk/6.0.202/Sdks/Microsoft.NET.Sdk/targets/Microsoft.PackageDependencyResolution.targets(267,5): error NETSDK1047: Assets file '/source/Pinta.Tools/obj/project.assets.json' doesn't have a target for 'net6.0/linux-x64'. Ensure that restore has run and that you have included 'net6.0' in the TargetFrameworks for your project. You may also need to include 'linux-x64' in your project's RuntimeIdentifiers. [/source/Pinta.Tools/Pinta.Tools.csproj]
  Pinta.Resources -> /source/Pinta.Resources/bin/Release/net6.0/Pinta.Resources.dll
  Pinta.Core -> /source/Pinta.Core/bin/Release/net6.0/Pinta.Core.dll
  Pinta.Gui.Widgets -> /source/Pinta.Gui.Widgets/bin/Release/net6.0/Pinta.Gui.Widgets.dll
  Pinta.Effects -> /source/Pinta.Effects/bin/Release/net6.0/linux-x64/Pinta.Effects.dll
  Pinta.Effects -> /app/
The command '/bin/sh -c dotnet publish -c release -o /app --no-restore --runtime linux-x64 --self-contained' returned a non-zero code: 1
```

... which leads to:

```
  Pinta.Resources -> /source/Pinta.Resources/bin/Release/net6.0/Pinta.Resources.dll
  Pinta.Core -> /source/Pinta.Core/bin/Release/net6.0/Pinta.Core.dll
  Pinta.Gui.Widgets -> /source/Pinta.Gui.Widgets/bin/Release/net6.0/Pinta.Gui.Widgets.dll
  Pinta.Tools -> /source/Pinta.Tools/bin/Release/net6.0/Pinta.Tools.dll
  Pinta.Tools -> /app/
  Pinta.Effects -> /source/Pinta.Effects/bin/Release/net6.0/Pinta.Effects.dll
  Pinta.Effects -> /app/
/usr/share/dotnet/sdk/6.0.202/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.RuntimeIdentifierInference.targets(150,5): error NETSDK1031: It is not supported to build or publish a self-contained application without specifying a RuntimeIdentifier. You must either specify a RuntimeIdentifier or set SelfContained to false. [/source/Pinta/Pinta.csproj]
The command '/bin/sh -c dotnet publish -c release -o /app --no-restore --self-contained' returned a non-zero code: 1
```

... which I'd say is a non-starter:

```console
$ app/Pinta 
A fatal error occurred. The required library libhostfxr.so could not be found.
If this is a self-contained application, that library should exist in [/home/stdedos/Documents/projects/ATracking/Pinta/app/].
If this is a framework-dependent application, install the runtime in the global location [/usr/share/dotnet] or use the DOTNET_ROOT environment variable to specify the runtime location or register the runtime location in [/etc/dotnet].

The .NET runtime can be found at:
  - https://aka.ms/dotnet-core-applaunch?missing_runtime=true&arch=x64&rid=ubuntu.20.04-x64&apphost_version=6.0.4
```

Unless you'd know how to include those dependencies as "another" debian file. "I feel" that no-one cares, but if there is something, let me know. One possibility, ofc, would be https://dot.net/v1/dotnet-install.sh and:
```console
$  ./dotnet-install.sh --dry-run
dotnet-install: Note that the intended use of this script is for Continuous Integration (CI) scenarios, where:
dotnet-install: - The SDK needs to be installed without user interaction and without admin rights.
dotnet-install: - The SDK installation doesn't need to persist across multiple CI runs.
dotnet-install: To set up a development environment or to run apps, use installers rather than this script. Visit https://dotnet.microsoft.com/download to get the installer.

dotnet-install: Payload URLs:
dotnet-install: URL #0 - aka.ms: https://dotnetcli.azureedge.net/dotnet/Sdk/6.0.300/dotnet-sdk-6.0.300-linux-x64.tar.gz
dotnet-install: Repeatable invocation: ./dotnet-install.sh --version "6.0.300" --install-dir "/home/stdedos/.dotnet" --architecture "x64" --os "linux"
```

Reply:
Maybe try removing the `--no-restore` argument?
For the macOS installer we do a self-contained publish that works fine (https://github.com/PintaProject/Pinta/blob/master/installer/macos/build_installer.sh#L18) so I'd try matching that. Maybe the `release` also needs to be capitalized?

Reply:
> Maybe try removing the `--no-restore` argument?

That's a Docker-specific optimization, recommended by Microsoft
https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/docker/building-net-docker-images?view=aspnetcore-6.0#the-dockerfile
This is not the one I copied it from, but it has been a while since I've read the specifics 😅

Reply:
The difference here Is that the publish command is being run with an explicit runtime identifier, vs the restore that happened when doing the normal build of Pinta, so I'm not sure if the restore can be skipped here?

Reply:
Actual source is: https://docs.microsoft.com/en-us/dotnet/core/docker/build-container?tabs=linux#create-the-dockerfile

But, however, you seem to be right - without the `--no-restore` flag, "everything seems to happen easily".
I wonder what is this behavior 😕 I might have to roll this back

Reply:
_There's so much complexity on "why things happen", which you can find https://github.com/PintaProject/Pinta/commit/ecf260151767a15edfe2ed7c0e831069cdf38d45 here and https://discord.com/channels/732297728826277939/732325020738519091/981673256249589780 here._

But it's fixed now

Reply:
The discord link doesn't work for me - could you paste the relevant info here?
There are issues with the --no-restore flag breaking normal builds of Pinta (see other comment)

I guess I still don't understand why you're needing to avoid a restore for the dotnet publish command...

Reply:
You probably need to join the server first: https://aka.ms/dotnet-discord

"It's better" on Docker terms to do it like that. It's right here:
> Actual source is: [docs.microsoft.com/en-us/dotnet/core/docker/build-container?tabs=linux#create-the-dockerfile](https://docs.microsoft.com/en-us/dotnet/core/docker/build-container?tabs=linux#create-the-dockerfile)
>
> In the preceding Dockerfile, the *.csproj files are copied and restored as distinct layers. When the docker build command builds an image, it uses a built-in cache. If the *.csproj files haven't changed since the docker build command last ran, the dotnet restore command doesn't need to run again. Instead, the built-in cache for the corresponding dotnet restore layer is reused. For more information, see [Best practices for writing Dockerfiles](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#leverage-build-cache).

However, I can just ignore that and do this the easy way - up to you.

Reply:
Yeah, I think keeping this to a minimal set of changes is a good starting point